apps/previewgenerator.sh

```diff
@@ -16,67 +16,209 @@ debug_mode
+            ;;
+
+            Movie)
+                occ_command config:system:set enabledPreviewProviders 9 --value="OC\\Preview\\Movie"
+            ;;
+
+            *)
+            ;;
+        esac
+    done 9< results
+    rm -f results
+    clear
+fi
+
+# Enable additional previews (most likely not working; remove the # to enable the specific preview)
+#occ_command config:system:set preview_libreoffice_path --value="/usr/bin/libreoffice"
+#occ_command config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\XBitmap"
+#occ_command config:system:set enabledPreviewProviders 8 --value="OC\\Preview\\Illustrator"
+#occ_command config:system:set enabledPreviewProviders 10 --value="OC\\Preview\\MSOffice2003"
+#occ_command config:system:set enabledPreviewProviders 11 --value="OC\\Preview\\MSOffice2007"
+#occ_command config:system:set enabledPreviewProviders 12 --value="OC\\Preview\\MSOfficeDoc"
+#occ_command config:system:set enabledPreviewProviders 13 --value="OC\\Preview\\OpenDocument"
+#occ_command config:system:set enabledPreviewProviders 14 --value="OC\\Preview\\PDF"
+#occ_command config:system:set enabledPreviewProviders 16 --value="OC\\Preview\\Postscript"
+#occ_command config:system:set enabledPreviewProviders 17 --value="OC\\Preview\\StarOffice"
+#occ_command config:system:set enabledPreviewProviders 20 --value="OC\\Preview\\Font"
+#occ_command config:system:set enabledPreviewProviders 21 --value="OC\\Preview\\HEIC"
+
+# check if the previewgenerator is installed and enabled
+if [ -d "$NC_APPS_PATH/previewgenerator" ]
+then
     # Set aspect ratio
-    occ_command config:app:set --value="32 64 1024"  previewgenerator squareSizes
-    occ_command config:app:set --value="64 128 1024" previewgenerator widthSizes
-    occ_command config:app:set --value="64 256 1024" previewgenerator heightSizes
+    occ_command config:app:set previewgenerator squareSizes --value="32 256"
+    occ_command config:app:set previewgenerator widthSizes  --value="256 384"
+    occ_command config:app:set previewgenerator heightSizes --value="256"
+    occ_command config:system:set preview_max_x --value="2048"
+    occ_command config:system:set preview_max_y --value="2048"
+    occ_command config:system:set jpeg_quality --value="60"
+    occ_command config:app:set preview jpeg_quality --value="60"
     
-    # Add crotab
-    crontab -u www-data -l | { cat; echo "@daily php -f $NCPATH/occ preview:pre-generate >> /var/log/previewgenerator.log"; } | crontab -u www-data -
+    msg_box "In the last step you can define a nextcloud-user for which you want to run the preview-generation. "
+    if [[ "yes" == $(ask_yes_or_no "So do you want to choose a nextcloud-user?") ]]
+    then
+        nextcloud_user=$(whiptail --inputbox "Enter the nextcloud-user for which you want to run the preview-generation" 10 30 3>&1 1>&2 2>&3)
+        export nextcloud_user
+        if [[ "yes" == $(ask_yes_or_no "Is this correct? $nextcloud_user") ]]
```

Comment:
@enoch85 btw: do we need this check? (Because what should happen if you press no?)

Reply:
Better safe than sorry. :)

Reply:
I will improve everything once I have time. Just commenting the obvious now.